### ISSUE TITLE
Add external source with ability to reference OSM id

### ISSUE THREAD

**Issue by HarelM** [IsraelHikingMap/Site#1353]
## Feature
In order to improve the ability to add external sources and to involve the community more we would like to start a prove of concept with גבים ושיטפונות.
The document is here:
https://docs.google.com/spreadsheets/d/1BbT762qaYilLnwGD7sJDo8sfs_Y5ExjvixS_Tl3rSas/edit?ts=5b3cda2d#gid=1535454640
There are just a few records there in order to test this POC.

**Comment by HarelM** [comment]
Ok, so I have implemented the major part of this and tested it to receive this (on my development machine):
![image](https://user-images.githubusercontent.com/3269297/98450928-c23c6800-2149-11eb-82ba-0ab13ca9f68d.png)
As can be seen the external source is there (can be seen by the cat icon).
There's a description already from OSM, so the description is not overwritten - probably the main problem here.
There's already a website connected to this POI so the Facebook source is not being used - I think I'm OK with that...
@amiad what would you expect to see in such case?

**Comment by amiad** [comment]
I think the information need to be after the description.
To example (Sorry for the poor design)
![new design](https://user-images.githubusercontent.com/402108/98451730-49d9a500-2151-11eb-90ec-27c7b559302c.jpg)

I know it might be technically complicated.

**Comment by amiad** [comment]
We need to add more information: update date and source.
מצב: מלאה (02/11/2020, [מקור](https://www.facebook.com/groups/myisrael100/permalink/4069040479777862/))

**Comment by HarelM** [comment]
Source will be problematic, but the icon can link to different places I guess if we set it this way in the sheets table.
I need to think how to properly implement this as the current merge algorithm doesn't add description if it's not empty in OSM, and also I don't want it to be accidently added to OSM if someone edits these POIs...
I thought of using the date field that's already there, but seems like it does not get updated, not sure why, I need to check.
Thanks for the detailed response!

**Comment by amiad** [comment]
The source can be in the icon but we will need to replace the icon for users click its. Maybe waterhole icon.
The update date cannot be in the regular date field because any edit of the POI will change it.

**Comment by HarelM** [comment]
@amiad Thanks for your time and thought.
The problem I'm currently facing is that in order to support this I need to do major changes in UI/UX and server code as opposed to what I thought in the first place.
The concept of "status" doesn't exists and opens up a lot of questions.
After further thought and talking to @zstadler we came up with a different approach and I would like to know what you think about it.
Here goes:
The site currently supports mapbox style json file.
Assuming there will be a script that creates and updates a geojson source with the data that exist in the file I think we can create a layer that will show the text of the description from the IHM sheet page on the map itself along side the date.
This layer can be shared and added to the overlay layers and users will be able to toggle it on and off.
Something like the following:
![image](https://user-images.githubusercontent.com/3269297/98470401-f9fceb80-21ed-11eb-8b0d-0fec6b8935f5.png)
![image](https://user-images.githubusercontent.com/3269297/98470364-c326d580-21ed-11eb-8d80-70e1e5c858f5.png)
If we open a repository and and some github actions I think it might be possible without a lot of hassle.
What do you think?

**Comment by HarelM** [comment]
This article is also interesting:
https://blog.mapbox.com/mapping-a-google-doc-spreadsheet-5d58b38b102b
That's also interesting in the same direction, seems we're not the first one to think about it.
https://gis.stackexchange.com/questions/140995/publishing-google-sheet-to-web-as-geojson-file

**Comment by amiad** [comment]
I understand that it is complicated to add information after the description. A new layer is an interesting idea but I like it less, it looks weird.
I will try to think of a beautiful solution that will also be easy to implement.

**Comment by HarelM** [comment]
OK, let me know what you come up with.
Please note that I did the mapping style very badly - you can play with the font size, color, halo etc to achieve a more esthetic solution.
I'll try to create a proof of concept tonight to let you see how it feels better.

**Comment by HarelM** [comment]
Check the following source for overlay address:
https://github.com/HarelM/IHMMapboxMapExample/blob/main/style.json
And checkout עין נמר.
![image](https://user-images.githubusercontent.com/3269297/98560576-91c30e00-22b0-11eb-8821-d6eb60e28dd5.png)
More styling can be applied, more points can be added.

**Comment by HarelM** [comment]
The following is an example repository with all the information how to achieve this.
https://github.com/HarelM/IHMMapboxMapExample
Please let me know if this can solve this issue.
@valleyofdawn this can be used to create a custom water taps layer if you want...

**Comment by amiad** [comment]
I less like the idea of a separate layer. I tried to think a lot about a other solution so I waited with my comment.
Maybe add separate POIs like the ascends from Jeepolog.

We can show different icons according to status of waterhole (full, semi-full or empty).

**Comment by HarelM** [comment]
Thanks for taking the time to think it over.
Adding a separate POI layer/category for data that exist in OSM will cause a POI "duplication" - since the points are already in OSM adding another layer with different points will create two points for every waterhole making a bad user experience. (Asends currently do not exists in OSM so there's no problem here, yet...).
The original solution I proposed was an external source like Ascends but then the description was not used since OSM has a description and thus the description from the external source was not used (this is how the merge algorithm works in the site - if it merges two points is will prefer OSM description).
I have also consulted with the OSM slack channel regarding the data that is available in the sheets and they told me that it should not be in OSM since it can change frequently (much like snow level in ski resorts), so creating a script that will update OSM entities is not recommended.
The bottom line is that I can't think of a better solution besides the custom layer.
In regards to what you said, I think a text describing the status is more intuitive to understand than a small icon. Adding a custom layer allows users who are interested in the data to see it while users who are not interested will not see it.
I'm sorry I don't have a better idea... :-(

**Comment by HarelM** [comment]
As a side note, creating a custom layer in order to see how many people are interested in this can be a good stating point as a proof of concept - it's relatively easy to create and update this (even manually at the beginning) in order to understand if this is useful or not for many users. If this catches up and many users are interested in this we might think how to improve this...

**Comment by amiad** [comment]
I have no other idea other than to add the information after the description.
Custom layer can be temporary solution butII'm afraid it will become permanent.

**Comment by HarelM** [comment]
As I said, adding it to the description is not advised by the OSM community...
I suggest to start with something. I think the layer idea is better than the current situation. Nothing is permanent in software :-)
Let me know if you need any help in creating a geojson from the google sheet and upload it to github.
I'll be closing this issue now as the proposed solution is external to this repository.

**Comment by zstadler** [comment]
Perhaps the [condition tag] (https://wiki.openstreetmap.org/wiki/Key:condition) can ce used to store the waterhole's condition in OSM, potentially with the use of the [check_date tag] (https://wiki.openstreetmap.org/wiki/Key:check_date).

**Comment by amiad** [comment]
I created GeoJSON (I used [this](https://www.convertcsv.com/csv-to-geojson.htm).
https://github.com/amiad/IHMGeoJSONWaterholes

I still have to design it.

**Comment by HarelM** [comment]
Cool! 
Note that I have created a single field for the text in order to style it easily. You can probably do it in google docs as a calculated field and have the geoJson and style.json use it. not mandatory of course, having it in multiple field can allow you to style the different text with different styles...
Let me know if you need any help :-)
